### PR TITLE
Feature integrate edit profile

### PR DESCRIPTION
**Pull request by sholabioku**
### What this PR does (required):
- lntegrate edit profile page

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

**Review by ezeilo-su — COMMENTED** [review]
Looks good! Please remember to always include a few screenshots alongside filling in the PR templates :smiley:

File: server/controllers/profile.js [hunk 1]
```diff
@@ -12,11 +12,14 @@ exports.editProfile = asyncHandler(async (req, res, next) => {
     res.status(404);
     throw new Error("User doesn't exist");
   }
-  user.profile.set(req.body);
-  const updatedUser = await user.save();
+
+  const profile = await Profile.findByIdAndUpdate(user.profile, req.body, {
+    new: true,
+  });
```
Comment (ezeilo-su) [inline server/controllers/profile.js:18]: :+1:

**Review by ezeilo-su — COMMENTED** [review]
Thank you for making the requested changes.

File: client/src/components/EditProfile/EditForm/useStyles.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles(() => ({
+  boxButton: {
+    textAlign: 'center',
+    paddingTop: '10px',
+  },
+  typography: {
+    fontWeight: 400,
+    textTransform: 'uppercase',
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditForm/useStyles.ts:10]: 👍🏽

File: client/src/components/EditProfile/EditForm/EditForm.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      description,
+      gender,
+      email,
+      address,
+      phoneNumber,
+      dateOfBirth,
+      availability,
+      photo,
+    }: {
+      firstName: string;
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditForm/EditForm.tsx:25]: Optional: You can use the Profile Interface here

File: client/src/components/EditProfile/EditMenu.tsx [hunk 4]
```diff
@@ -42,12 +42,81 @@ function a11yProps(index: number) {
...
+      description,
+      gender,
+      email,
+      address,
+      phoneNumber,
+      dateOfBirth,
+      availability,
+      photo,
+    }: {
+      firstName: string;
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditMenu.tsx:66]: You can use the Profile interface here too

File: client/src/components/EditProfile/EditMenu.tsx [hunk 5]
```diff
@@ -42,12 +42,81 @@ function a11yProps(index: number) {
...
+      address: string;
+      phoneNumber: string;
+      dateOfBirth: string;
+      availability: string;
+      photo: string;
+    },
+    {
+      setSubmitting,
+    }: FormikHelpers<{
+      firstName: string;
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditMenu.tsx:80]: Here as well

File: client/src/components/FormsUI/Button/index.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  children?: React.ReactNode;
+  onClick?: (values: any) => void;
+  type: string;
+  size: 'large' | 'medium' | 'small';
+  variant: 'text' | 'contained' | 'outlined';
+  className?: string;
+  fullWidth?: boolean;
+}
+
+export const ButtonWrapper: React.FC<Props> = ({ color, size, variant, fullWidth, children }) => {
```
Comment (rajivtitus) [inline client/src/components/FormsUI/Button/index.tsx:16]: We probably didn't need to componentize this, as we are returning only a Button :)

File: client/src/components/FormsUI/Select/index.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { TextField, MenuItem } from '@material-ui/core';
+import { useField, useFormikContext } from 'formik';
+
+export const SelectWrapper = ({ name, options, ...otherProps }: any): JSX.Element => {
```
Comment (rajivtitus) [inline client/src/components/FormsUI/Select/index.tsx:5]: Please avoid using 'any' as a type

File: client/src/components/EditProfile/EditForm/EditForm.tsx [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+});
+
+const EditForm = ({ handleSubmit }: Props): JSX.Element => {
+  const classes = useStyles();
+  const options = ['Male', 'Female', 'Other'];
+  return (
+    <Box>
+      <Formik initialValues={initialFormState} validationSchema={formValidation} onSubmit={handleSubmit}>
+        {({ handleSubmit }) => (
+          <Form onSubmit={handleSubmit}>
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditForm/EditForm.tsx:87]: Nice work using Formik's form component here 👍🏽

**Review by rajivtitus — APPROVED** [review]
Good work making the requested changes from the UI PR within this PR 👍🏽 

Note: For IN PRs please consider adding a few screenshots/videos to show the feature in action!

**Pull request merged**